front/src/modules/auth/states/currentUserState.ts

```diff
@@ -1,8 +1,32 @@
 import { atom } from 'recoil';
 
-import { GetCurrentUserQuery } from '~/generated/graphql';
+import {
+  User,
+  UserSettings,
+  Workspace,
+  WorkspaceMember,
+} from '~/generated/graphql';
 
-export type CurrentUser = GetCurrentUserQuery['currentUser'];
+export type CurrentUser = Pick<
```

Comment:
@gitstart-twenty as soon as we are out of graphql layer, we precisely type our object based on what we need using Pick<GraphQLObject>. This makes sure the app/component has what it needs to run, no matter what happens in at the query level.